src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -552,6 +549,8 @@ class LazyMobXTableStore<T> {
 
         this._page = 0;
         this.itemsPerPage = lazyMobXTableProps.initialItemsPerPage || 50;
+
+        reaction(()=>this.displayData.length, ()=>{ this.page = this.page; /* trigger recomputation of page when data length changes */});
```

Comment:
this line seems potentially like a mistake
  this.page = this.page?

Reply:
no its right - in `set page` it clamps the input value, and when the display data length changes it needs to be potentially clamped. So this just triggers that recomputation, as it says in the comment